### PR TITLE
✨ Onboard rsETH on Aave V3 Ethereum

### PR DESCRIPTION
**Pull request by MartinGbz**
## Recap

- Add rsETH to Aave V3 Ethereum Pool
- Create a new "rsETH E-Mode"
- Setup rsETH, wstETH and ETHx to the rsETH E-Mode
- Update wstETH and ETHx borrow caps

### CODE REVIEW

File: src/20241104_AaveV3Ethereum_OnboardRsETHToAaveV3Ethereum/OnboardRsETHToAaveV3Ethereum.md
Comment (Matt-Graham-TL) [inline src/20241104_AaveV3Ethereum_OnboardRsETHToAaveV3Ethereum/OnboardRsETHToAaveV3Ethereum.md:1]: Would it be possible to provide details on the eMode being created for visibility ?

Comment (MartinGbz) [reply src/20241104_AaveV3Ethereum_OnboardRsETHToAaveV3Ethereum/OnboardRsETHToAaveV3Ethereum.md:1]: You can find the emode configuration here: https://governance.aave.com/t/arfc-add-rseth-to-aave-v3-ethereum/17696#p-44683-rsethwstethethx-8

fyi: the forum post has just been cleaned

File: src/20241104_AaveV3Ethereum_OnboardRsETHToAaveV3Ethereum/AaveV3Ethereum_OnboardRsETHToAaveV3Ethereum_20241104.sol [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+  {
+    IAaveV3ConfigEngine.EModeCategoryUpdate[]
+      memory eModeUpdates = new IAaveV3ConfigEngine.EModeCategoryUpdate[](1);
+
+    eModeUpdates[0] = IAaveV3ConfigEngine.EModeCategoryUpdate({
+      eModeCategory: 2,
+      ltv: 92_50,
+      liqThreshold: 94_50,
+      liqBonus: 1_00,
+      label: 'rsETH'
```
Comment (brotherlymite) [inline src/20241104_AaveV3Ethereum_OnboardRsETHToAaveV3Ethereum/AaveV3Ethereum_OnboardRsETHToAaveV3Ethereum_20241104.sol:65]:
```suggestion
      label: 'rsETH LST main'
```
To be consistent, I would add the label as `rsETH LST main`

File: src/20241104_AaveV3Ethereum_OnboardRsETHToAaveV3Ethereum/OnboardRsETHToAaveV3Ethereum.md
Comment (brotherlymite) [reply src/20241104_AaveV3Ethereum_OnboardRsETHToAaveV3Ethereum/OnboardRsETHToAaveV3Ethereum.md:1]: @MartinGbz I think better to have it on AIP too, also I would include the cap increase changes too

Comment (MartinGbz) [reply src/20241104_AaveV3Ethereum_OnboardRsETHToAaveV3Ethereum/OnboardRsETHToAaveV3Ethereum.md:1]: What do you mean exactly?
From what I see, there is no other infos/data on the E-Mode except the table (that is already in this PR)

Comment (MartinGbz) [reply src/20241104_AaveV3Ethereum_OnboardRsETHToAaveV3Ethereum/OnboardRsETHToAaveV3Ethereum.md:1]: Yes indeed! I confused the md file generated by the test with the AIP md file.
All good now

File: src/20241104_AaveV3Ethereum_OnboardRsETHToAaveV3Ethereum/AaveV3Ethereum_OnboardRsETHToAaveV3Ethereum_20241104.sol (hunk 1 above)
Comment (MartinGbz) [reply src/20241104_AaveV3Ethereum_OnboardRsETHToAaveV3Ethereum/AaveV3Ethereum_OnboardRsETHToAaveV3Ethereum_20241104.sol]: I preferred to rename it has "rsETH-Correlated", it makes more sens here (+ it has been renamed also liked this in the forum post)

File: diffs/AaveV3Ethereum_OnboardRsETHToAaveV3Ethereum_20241104_before_AaveV3Ethereum_OnboardRsETHToAaveV3Ethereum_20241104_after.md [hunk 2]
```diff
@@ -0,0 +1,172 @@
...
+| borrowCap | 320 ETHx | 5,000 ETHx |
+
+
+## Emodes changed
+
+### EMode: ETH correlated(id: 1)
+
+
+
+### EMode: rsETH(id: 2)
```
Comment (ianflexa) [inline diffs/AaveV3Ethereum_OnboardRsETHToAaveV3Ethereum_20241104_before_AaveV3Ethereum_OnboardRsETHToAaveV3Ethereum_20241104_after.md:73]:
```suggestion
### EMode: rsETH-Correlated(id: 2)
```

File: diffs/AaveV3Ethereum_OnboardRsETHToAaveV3Ethereum_20241104_before_AaveV3Ethereum_OnboardRsETHToAaveV3Ethereum_20241104_after.md [hunk 3]
```diff
@@ -0,0 +1,172 @@
...
+
+| description | value before | value after |
+| --- | --- | --- |
+| eMode.label | - | rsETH |
```
Comment (ianflexa) [inline diffs/AaveV3Ethereum_OnboardRsETHToAaveV3Ethereum_20241104_before_AaveV3Ethereum_OnboardRsETHToAaveV3Ethereum_20241104_after.md:77]:
```suggestion
| eMode.label | - | rsETH-Correlated |
```

File: diffs/AaveV3Ethereum_OnboardRsETHToAaveV3Ethereum_20241104_before_AaveV3Ethereum_OnboardRsETHToAaveV3Ethereum_20241104_after.md [hunk 4]
```diff
@@ -0,0 +1,172 @@
...
+```json
+{
+  "eModes": {
+    "2": {
+      "from": null,
+      "to": {
+        "borrowableBitmap": 2147483650,
+        "collateralBitmap": 68719476736,
+        "eModeCategory": 2,
+        "label": "rsETH",
```
Comment (ianflexa) [inline diffs/AaveV3Ethereum_OnboardRsETHToAaveV3Ethereum_20241104_before_AaveV3Ethereum_OnboardRsETHToAaveV3Ethereum_20241104_after.md:96]:
```suggestion
        "label": "rsETH-Correlated",
```

File: src/20241104_AaveV3Ethereum_OnboardRsETHToAaveV3Ethereum/AaveV3Ethereum_OnboardRsETHToAaveV3Ethereum_20241104.sol (hunk 1 above)
Comment (MartinGbz) [reply src/20241104_AaveV3Ethereum_OnboardRsETHToAaveV3Ethereum/AaveV3Ethereum_OnboardRsETHToAaveV3Ethereum_20241104.sol]: Indeed, there's only 1 "ETH Correlated" e-mode, the other ones use the format you proposed. So yes it makes more sens to follow this format okay 👍

**Comment by MartinGbz** [comment]
@brotherlymite I've fixed emode label
And I also update emode category id from `2` to `3` (as the recently merged sUSDe emode already use category 2)

**Review by brotherlymite — APPROVED** [review]

**Review by ianflexa — APPROVED** [review]

**Pull request merged**